k2/csrc/fsa.h

```diff
@@ -1,15 +1,37 @@
-#ifndef K2_FSA_H_
-#define K2_FSA_H_
+// k2/csrc/fsa.h
+
+// Copyright     2020  Daniel Povey
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef K2_CSRC_FSA_H_
+#define K2_CSRC_FSA_H_
+
+#include <cstdint>
 #include <vector>
-#include <stdint.h>
 
 namespace k2 {
 
 enum {
-  kFinalSymbol = -1,   /* final-costs are represented as arcs with kFinalSymbol as their
-                          label, to the final state (see Fst::final_state). */
-  kEpsilon = 0         /* Epsilon, which means "no symbol", is numbered zero,
-                        * like in OpenFst. */
+  kFinalSymbol = -1,   //!< final-costs are represented as arcs with
```

Comment:
This is Doxygen-style formatting `(//!)`.   My feeling is we could forgo the Doxygen, as these days people use Sphinx more?

Reply:
I will change it to `//`

`cpplint` issues warnings if you use `/**/` for multiline comment.

Reply:
.. BTW, I'm still using the @param [in] foo function-parameter documentation because I'm used to it and I feel it's readable as text.  I'm open to changing it though, to anything that's clear and consistent.

Reply:
`//!<`  is replaced with `//`.

I find that `@param [in]` is more readable.